src/Helpers/Hooks.js

```diff
@@ -32,3 +33,11 @@ export const useNotification = (config) => {
     return addNotification;
 };
 
+export const useClearNotifications = () => {
```

Comment:
we already have `useNotification` that handles the `addNotification`. Why do you need to create a seperate hook to handle the `clearNotification` ?

Reply:
Sometimes we need to clear notif before the 8 second dismiss period, for example when user clicks Download report, preparing report is shown, then when report is ready this notif is cleared and downloading report is shown. I could use the raw clearNotifications from FEC, but with this hook I don't have to add useDispatch to components which don't have it.

Reply:
I see the use case here, what I'm suggesting here is instead of creating a separate hook clearing the notification queue to include it in `useNotification` 



What can we achieve? 
1. one hook/place handling the notification
2. import only  `useNotification` and use ` const [addNotification, clearNotifications] = useNotification();`

Reply:
@katerinapat This gave me an idea, why even call `clearNotifications`, which clears all notifications? Here we only want to remove one particular notification (without affecting any other possible notifications (unlikely but possible)), so why not Use `removeNotification(notif)` which only removes one particular `notif`.

```js
const  { addNotification, removeNotification } = useNotification(); // object destructuring, not array destructuring to force us to use consistent variable names

const reportInitNotif = addNotification({ title: ..., body: ...}); // Report is being built...
// wait until report is built
removeNotification(reportInitNotif); // only removes this one particular notif
addNotification({ title: ..., body: ...}); // Downloading report
```

OR

```js
const addNotification = useNotification();

const reportInitNotif = addNotification({ title: ..., body: ...}); // addNotification will return notification object with remove() method
// wait until report is built
reportInitNotif.remove(); // only removes this one particular notif
addNotification({ title: ..., body: ...}); // Downloading report
```

Reply:
That's an interesting point but why would you have to define/call all the these steps  on every single file/function to clear the previous notification?
 
Let me give you an example about **declarative vs. imperative thinking**. 
I will use the example of asking an espresso to explain the difference  :smile: 

#### imperative thinking
1. go to the kitchen
2. turn on the coffee machine
3. press espress
3. get a mug 
4. fill the mug 

#### declarative thinking
1.  I want an espresso without sugar or milk (basically I don't care **_how_** is going to be prepared but I rather care about the **result**)

#### How does this apply to our example?
Your example is more imperative, we'll have to **describe every time** we use the `addNotification` to clear the previous one. 
Would be possible to make it more declarative? For example, when we ask for (espresso) `addNotification` to have the result we desire. in this case, `addNotification` should know the previous notification and remove it!

Reply:
Fixed :)